lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -446,40 +446,22 @@ def _get(
                    }             }           }               }
                   }             }           }               }
         """
-        mtable = aliased(Metadata)
-        stable = aliased(Metadata)
-        gtable = aliased(Metadata)
-        utable = aliased(Metadata)
         aliases = {
-            Metadata.METALOG: mtable,
-            Metadata.SERVER: stable,
-            Metadata.GLOBAL: gtable,
-            Metadata.USER: utable,
+            Metadata.METALOG: aliased(Metadata),
+            Metadata.SERVER: aliased(Metadata),
+            Metadata.GLOBAL: aliased(Metadata),
+            Metadata.USER: aliased(Metadata),
         }
-        query = (
-            Database.db_session.query(Dataset)
-            .outerjoin(
-                mtable,
-                and_(mtable.dataset_ref == Dataset.id, mtable.key == Metadata.METALOG),
-            )
-            .outerjoin(
-                stable,
-                and_(stable.dataset_ref == Dataset.id, stable.key == Metadata.SERVER),
-            )
-            .outerjoin(
-                gtable,
-                and_(gtable.dataset_ref == Dataset.id, gtable.key == Metadata.GLOBAL),
-            )
-        )
-        if auth_id:
-            query = query.outerjoin(
-                utable,
-                and_(
-                    utable.dataset_ref == Dataset.id,
-                    utable.key == Metadata.USER,
-                    utable.user_id == auth_id,
-                ),
-            )
+        query = Database.db_session.query(Dataset)
+        for key, table in aliases.items():
+            terms = [table.dataset_ref == Dataset.id, table.key == key]
+            if key == Metadata.USER:
+                if auth_id:
+                    terms.append(table.user_id == auth_id)
+                else:
+                    continue
+            current_app.logger.info("Adding JOIN {}", key)
+            query = query.outerjoin(table, and_(*terms))
```

Comment:
The use of `continue` here is very clever.  So clever in fact that I would suggest a small restructuring of the code.  As it is, you're sort of "burying the lead" -- the point is that, if there's a `USER` key and no `auth_id`, then we skip adding the term altogether, but we've got to follow three levels of indentation and get into the `else` clause to reach that.  I think the following gets us there more clearly:
```python
            if key == Metadata.USER:
                if not auth_id:
                    continue
                terms.append(table.user_id == auth_id)

            current_app.logger.info("Adding JOIN {}", key)
            query = query.outerjoin(table, and_(*terms))
```
It's a small change, but now it's more obvious that the `not auth_id` case is the exception which skips the rest of the stuff; if we don't skip, then we do the rest of the `USER` special case, then follow the normal path.

Ditto for the analogous code in the unit tests.

Reply:
Maybe ... not sure it's worth the effort, but I need to remove the dangling debug code anyway.